lib/components/Scene/Scene.tsx

```diff
@@ -764,16 +764,15 @@ export const Session: React.FC<IProps> = (props) => {
   function renderCharacterCards() {
     const everyone = sessionManager.computed.everyone;
     const characters = sessionCharactersManager.state.characterSheets;
-    const playersWithCharacterSheets = Object.keys(characters)
-      .map((characterId) => {
-        const playerMatch = everyone.find(
-          (player) => player.id === characterId
-        ) as IPlayer;
+    const playersWithCharacterSheets = everyone
+      .map((player) => {
+        const characterSheetMatch = characters[player.id];
         return {
-          ...playerMatch,
-          characterSheet: characters[characterId],
+          ...player,
+          characterSheet: characterSheetMatch,
         };
```

Comment:
The old code was going through the `characters` dictionary, and matching it with players. 
It should have been the other way around, in case character are out of sync. 
It makes more sense to go through players, match them with their character and filter out players without characters (new code)